### PR TITLE
[atv.at] fix initial video extraction

### PR DESCRIPTION
**Pull request by NeroBurner**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) - No, just updated one line (regex-string)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

ATV.at did change the HTML structure. They moved the json data into a
variable. Now the extractor looks for the variable instead

Fixes: https://github.com/rg3/youtube-dl/issues/18041

### CODE REVIEW

File: youtube_dl/extractor/atvat.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def _real_extract(self, url):
         display_id = self._match_id(url)
         webpage = self._download_webpage(url, display_id)
         video_data = self._parse_json(unescapeHTML(self._search_regex(
-            r'class="[^"]*jsb_video/FlashPlayer[^"]*"[^>]+data-jsb="([^"]+)"',
+            r'var flashPlayerOptions[ ]*=[ ]*"([^\n]+)";',
```
Comment (dstftw) [inline youtube_dl/extractor/atvat.py]: 1. Do not remove the old regex.
2. `\s*` for matching whitespace.
3. When matching between quotes - match any non quote character.
4. Remove useless `;`.

Comment (NeroBurner) [reply youtube_dl/extractor/atvat.py]: fixed

**Comment by NeroBurner** [comment]
@dstftw fixed, please recheck `pending-fixes` label

File: youtube_dl/extractor/atvat.py [hunk 2]
```diff
@@ -28,7 +28,8 @@ def _real_extract(self, url):
         display_id = self._match_id(url)
         webpage = self._download_webpage(url, display_id)
         video_data = self._parse_json(unescapeHTML(self._search_regex(
-            r'class="[^"]*jsb_video/FlashPlayer[^"]*"[^>]+data-jsb="([^"]+)"',
+            #r'class="[^"]*jsb_video/FlashPlayer[^"]*"[^>]+data-jsb="([^"]+)"',
```
Comment (dstftw) [inline youtube_dl/extractor/atvat.py]: I mean you must pass both regexes.

File: youtube_dl/extractor/atvat.py [hunk 3]
```diff
@@ -28,7 +28,8 @@ def _real_extract(self, url):
...
+            r'var\s*flashPlayerOptions\s*=\s*"([^"]+)"',
```
Comment (dstftw) [inline youtube_dl/extractor/atvat.py]: Should not match `varflashPlayerOptions...`.

**Comment by NeroBurner** [comment]
@dstftw please review

**Pull request merged**